patches/server/0880-Allow-usage-of-MiniMessage-in-server-configuration.patch

```diff
@@ -0,0 +1,410 @@
++                getString("messages.kick.connection-throttle", "Connection throttled! Please wait before reconnecting.")
++            );
+     }
+ 
+-    public static String noPermissionMessage = "&cI'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.";
++    public static Component noPermissionMessage;
+     private static void noPermissionMessage() {
+-        noPermissionMessage = ChatColor.translateAlternateColorCodes('&', getString("messages.no-permission", noPermissionMessage));
++        if (configurationUsesMiniMessage)
++            noPermissionMessage = miniMessage().deserialize(
++                getString("messages.no-permission", "<red>I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.")
++            );
++
++        else
++            noPermissionMessage = LegacyComponentSerializer.legacyAmpersand().deserialize(
++                getString("messages.no-permission", "&cI'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.")
++            );
+     }
+ 
+     private static void savePlayerData() {
+@@ -552,13 +593,23 @@ public class PaperConfig {
+         }
+     }
+ 
+-    public static String kickMessage;
++    public static Component kickMessage;
+     public static PacketLimit allPacketsLimit;
+     public static java.util.Map<Class<? extends net.minecraft.network.protocol.Packet<?>>, PacketLimit> packetSpecificLimits = new java.util.HashMap<>();
+ 
+     private static void packetLimiter() {
+         packetSpecificLimits.clear();
+-        kickMessage = org.bukkit.ChatColor.translateAlternateColorCodes('&', getString("settings.packet-limiter.kick-message", "&cSent too many packets"));
++
++        if (configurationUsesMiniMessage)
++            kickMessage = miniMessage().deserialize(
++                getString("settings.packet-limiter.kick-message", "<red>Sent too many packets")
++            );
++
++        else
++            kickMessage = legacyAmpersand().deserialize(
++                getString("settings.packet-limiter.kick-message", "&cSent too many packets")
++            );
++
+         allPacketsLimit = new PacketLimit(
+             getDouble("settings.packet-limiter.limits.all.interval", 7.0),
+             getDouble("settings.packet-limiter.limits.all.max-packet-rate", 500.0),
+@@ -684,4 +735,9 @@ public class PaperConfig {
+     private static void useDimensionTypeForCustomSpawners() {
+         useDimensionTypeForCustomSpawners = getBoolean("settings.use-dimension-type-for-custom-spawners", false);
+     }
++
++    public static boolean configurationUsesMiniMessage;
++    private static void configurationUsesMiniMessage() {
++        configurationUsesMiniMessage = getBoolean("settings.configuration-uses-minimessage", version > 25);
```

Comment:
Not sure I like this. How about instead, we check if the message contains ampersand, then revert to legacy, else default to MiniMessage

Reply:
Just checking for usage of & could cause issues incase someone is using it other then formatting, having a config option is much more clear and leaves no room for it to get confused.

Reply:
I don't think there should be a config option at all. And if the config version is < 26, all the messages should be force updated to minimessage.

Reply:
Okay, I guess I will just force all of the messages up to MiniMessage then.